numpy/linalg/umath_linalg.c.src

```diff
@@ -3233,6 +3233,28 @@ static void
     set_fp_invalid_or_clear(error_occurred);
 }
 
+static void
+@TYPE@_qr(char **args, npy_intp const *dimensions, npy_intp const *steps,
+          void *NPY_UNUSED(func))
+{
+    GELSD_PARAMS_t params;
+    int error_occurred = get_fp_invalid_and_clear();
+    fortran_int n, m;
+    fortran_int excess;
+
+    INIT_OUTER_LOOP_7
+
+    m = (fortran_int)dimensions[0];
+    n = (fortran_int)dimensions[1];
+    excess = m - n;
+
+    if (init_@lapack_func@(&params, m, n, nrhs)) {
+        LINEARIZE_DATA_t a_in, q_out, r_out;
+
+    }
```

Comment:
As of now this is empty. I will this in the upcoming commits. `a_in` is the square matrix (should we allow non-square matrices as well? QR decomposition is defined only for square matrix). `q_out` is the Q component of the decomposition and `r_out` is the R component of the decomposition.